transports/quic/src/provider/async_std.rs

```diff
@@ -36,6 +36,7 @@ pub struct Provider;
 
 impl super::Provider for Provider {
     type IfWatcher = if_watch::smol::IfWatcher;
+    type UdpSocket = async_std::net::UdpSocket;
```

Comment:
Looks good but this leaks the underlying tokio/async-std runtime. If that's okay, I don't mind merging it.